src/parsers/time-parser.js

```diff
@@ -31,8 +28,15 @@ export default class TimeParser {
       const start = date.start ? Number(date.start.date()) : null;
       const end = date.end ? Number(date.end.date()) : null;
       const extractedText = date.text;
-      const processedText = text.substr(0, date.index) +
-        text.substr(date.index + extractedText.length);
+      let beforeText = text.substr(0, date.index).trim();
+      const afterText = text.substr(date.index + extractedText.length).trim();
+
+      // temporary fix for https://github.com/wanasit/chrono/issues/152
+      if (/ (?:on|by)$/.test(beforeText)) {
```

Comment:
yeah well I made this one on prupose to make sure it was 3 characters ;) but well slice(0, -3) works if the space is missing so I guess I can do that.